### ISSUE TITLE
"Calling Accessor on non-internal object" spams logs at an incredible rate

### ISSUE THREAD

**Issue by stevekuznetsov** [kubernetes/kubernetes#26570]
Running conformance extended tests in Origin (unclear if this shows up as badly on normal operation, or in Kube alone), I'm seeing hundreds (100-400) of lines of this *per second*:
```
I0531 15:43:58.196056   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.HorizontalPodAutoscalerList
I0531 15:43:58.196168   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.IngressList
I0531 15:43:58.197782   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.HorizontalPodAutoscalerList
I0531 15:43:58.197934   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.IngressList
I0531 15:43:58.200721   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.JobList
I0531 15:43:58.203942   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.JobList
I0531 15:43:58.205782   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.DaemonSetList
I0531 15:43:58.207391   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.DaemonSetList
I0531 15:43:58.208995   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.JobList
I0531 15:43:58.209733   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.DeploymentList
I0531 15:43:58.211183   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.JobList
I0531 15:43:58.212003   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.DeploymentList
I0531 15:43:58.213117   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.ReplicaSetList
I0531 15:43:58.214085   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.HorizontalPodAutoscalerList
I0531 15:43:58.215305   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.ReplicaSetList
I0531 15:43:58.216069   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.HorizontalPodAutoscalerList
I0531 15:43:58.217221   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.ThirdPartyResourceList
I0531 15:43:58.218224   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.IngressList
I0531 15:43:58.219020   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.ThirdPartyResourceList
I0531 15:43:58.220285   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.IngressList
I0531 15:43:58.221188   10585 meta.go:46] Calling Accessor on non-internal object: *api.ConfigMapList
I0531 15:43:58.222132   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.JobList
I0531 15:43:58.223344   10585 meta.go:46] Calling Accessor on non-internal object: *api.ConfigMapList
I0531 15:43:58.223925   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.JobList
I0531 15:43:58.225645   10585 meta.go:46] Calling Accessor on non-internal object: *api.EndpointsList
I0531 15:43:58.225761   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.ReplicaSetList
I0531 15:43:58.227636   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.ReplicaSetList
I0531 15:43:58.227829   10585 meta.go:46] Calling Accessor on non-internal object: *api.EndpointsList
I0531 15:43:58.229273   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.ThirdPartyResourceList
I0531 15:43:58.229584   10585 meta.go:46] Calling Accessor on non-internal object: *api.EventList
I0531 15:43:58.230709   10585 meta.go:46] Calling Accessor on non-internal object: *extensions.ThirdPartyResourceList
I0531 15:43:58.230985   10585 meta.go:46] Calling Accessor on non-internal object: *api.EventList
I0531 15:43:58.233851   10585 meta.go:46] Calling Accessor on non-internal object: *api.LimitRangeList
I0531 15:43:58.234365   10585 meta.go:46] Calling Accessor on non-internal object: *api.ConfigMapList
I0531 15:43:58.237103   10585 meta.go:46] Calling Accessor on non-internal object: *api.ConfigMapList
I0531 15:43:58.237187   10585 meta.go:46] Calling Accessor on non-internal object: *api.LimitRangeList
I0531 15:43:58.239129   10585 meta.go:46] Calling Accessor on non-internal object: *api.EndpointsList
I0531 15:43:58.239333   10585 meta.go:46] Calling Accessor on non-internal object: *api.PersistentVolumeClaimList
I0531 15:43:58.241119   10585 meta.go:46] Calling Accessor on non-internal object: *api.EndpointsList
I0531 15:43:58.241321   10585 meta.go:46] Calling Accessor on non-internal object: *api.PersistentVolumeClaimList
I0531 15:43:58.245597   10585 meta.go:46] Calling Accessor on non-internal object: *api.PodList
I0531 15:43:58.246016   10585 meta.go:46] Calling Accessor on non-internal object: *api.EventList
I0531 15:43:58.248573   10585 meta.go:46] Calling Accessor on non-internal object: *api.EventList
I0531 15:43:58.254200   10585 meta.go:46] Calling Accessor on non-internal object: *api.LimitRangeList
```

Seems to be a `V(4)` ... but perhaps should be rate-limited?

**Comment by lavalamp** [comment]
This means you're hitting a slow path and it should be fixed. @smarterclayton

**Comment by lavalamp** [comment]
What version were you running at? Is this reproducible at head?

**Comment by stevekuznetsov** [comment]
We figured out this was due to the `ListMeta` not implementing the correct `Accessor`... I think @smarterclayton has a pull to address it.

**Comment by smarterclayton** [comment]
This was fixed and merged in 1.3 by
https://github.com/kubernetes/kubernetes/pull/27168

On Jun 29, 2016, at 4:31 PM, Steve Kuznetsov <notifications@github.com>
wrote:

We figured out this was due to the ListMeta not implementing the correct
Accessor... I think @smarterclayton <https://github.com/smarterclayton> has
a pull to address it.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/kubernetes/issues/26570#issuecomment-229519918>,
or mute the thread
<https://github.com/notifications/unsubscribe/ABG_p-0i8UnC-ZarVzjZJe60O1K36wNLks5qQwBcgaJpZM4IqrQo>
.
